uring/jvm/src/main/scala/fs2/io/uring/net/UringSocketGroup.scala

```diff
@@ -0,0 +1,217 @@
+
+                    bufLength.writeInt(buf.capacity())
+
+                    // We accept a connection, we write the remote address on the buf and we get the clientFd
+                    val accept =
+                      Resource.eval(F.delay(println("[SERVER] accepting connection..."))) *>
+                        ring
+                          .bracket(
+                            op = IORING_OP_ACCEPT,
+                            fd = linuxSocket.fd(),
+                            bufferAddress = buf.memoryAddress(),
+                            offset = bufLength.memoryAddress()
+                          )(closeSocket(ring, _))
+                          .mapK {
+                            new cats.~>[IO, IO] {
+                              def apply[A](ioa: IO[A]) = ioa.debug()
+                            }
+                          }
+                          .mapK(LiftIO.liftK)
+
+                    // We read the address from the buf and we convert it to SocketAddress
+                    val convert: F[SocketAddress[IpAddress]] =
+                      F.delay(
+                        println(
+                          "[SERVER] getting the address in memory and converting it to SocketAddress..."
+                        )
+                      ) *>
+                        /* TODO: Merge the next two steps in one:  F.delay (SocketAddress.fromInetSocketAddress(readIpv(buf.memoryAddress(), isIpv6))) */
+                        F.delay {
+                          val netRemoteAddress: InetSocketAddress =
+                            readIpv(buf.memoryAddress(), isIpv6)
+                          println(
+                            s"[SERVER] Read IP address from buffer: ${netRemoteAddress.getHostString()}"
+                          )
+                          netRemoteAddress
+                        }.flatMap { netRemoteAddress =>
+                          F.delay {
+                            println(
+                              s"[SERVER] converted and found inetSocketAddress: $netRemoteAddress"
+                            )
+                            SocketAddress.fromInetSocketAddress(netRemoteAddress)
+                          }
+                        }
+
+                    accept
+                      .flatMap { clientFd =>
+                        Resource.eval(convert).flatMap { remoteAddress =>
+                          UringSocket(ring, UringLinuxSocket(clientFd), clientFd, remoteAddress)
+                        }
+                      }
+                      .attempt
+                      .map(_.toOption)
+                  }.repeat
+              }
+
+          }
+
+      } yield (localAddress, sockets.unNone)
+    }
+
+  private[this] def openSocket(
+      ring: Uring,
+      ipv6: Boolean
+  ): Resource[F, UringLinuxSocket] =
+    Resource.make[F, UringLinuxSocket](F.delay(UringLinuxSocket.newSocketStream(ipv6)))(
+      linuxSocket => closeSocket(ring, linuxSocket.fd()).to
+    )
```

Comment:
This is fine for now but actually we can also open sockets with io_uring and avoid the syscall.

https://github.com/armanbilge/fs2-io_uring/blob/cc293e400d4f9572566474405a4a07898ef32ae3/uring/src/main/scala/fs2/io/uring/net/UringSocketGroup.scala#L117-L121